Framework/jmaqs-webservices/src/test/java/com/magenic/jmaqs/webservices/WebServiceDriverManagerUnitTest.java

```diff
@@ -0,0 +1,77 @@
+ */
+public class WebServiceDriverManagerUnitTest extends BaseTest {
+  @Test(groups = TestCategories.WebService)
+  public void getWebServiceDriverWithSupplierTest() throws URISyntaxException {
+    WebServiceDriver webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+    webServiceDriver.setHttpClient(webServiceDriver.getHttpClient(MediaType.APP_JSON.getMediaTypeString()));
+
+    WebServiceDriverManager driverManager = new WebServiceDriverManager(
+        () -> webServiceDriver.getHttpClient(MediaType.APP_JSON.getMediaTypeString()),
+        this.getTestObject());
+
+    Assert.assertNotNull(driverManager.getWebServiceDriver(), "Expected Web Service Driver to not be null");
+  }
+
+  @Test(groups = TestCategories.WebService)
+  public void getWebServiceDriverTest() throws URISyntaxException {
+    WebServiceDriver webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+    WebServiceDriverManager driverManager = new WebServiceDriverManager(webServiceDriver, this.getTestObject());
+
+    Assert.assertNotNull(driverManager.getWebServiceDriver(), "Expected Web Service Driver to not be null");
+  }
+
+  @Test(groups = TestCategories.WebService)
+  public void getWebServiceDriverNullDriver() throws URISyntaxException {
+    WebServiceDriver webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+    WebServiceDriverManager driverManager = new WebServiceDriverManager(
+        () -> webServiceDriver.getHttpClient(MediaType.APP_JSON.getMediaTypeString()),
+        this.getTestObject());
+
+    // Set the Driver to be null then check Get Web Service Driver creates default Driver.
+    driverManager.overrideDriver(null);
+    Assert.assertNotNull(driverManager.getWebServiceDriver(), "Expected Default Web Service Driver to be created.");
+  }
+
+  @Test(groups = TestCategories.WebService)
+  public void overrideWebServiceDriverTest() throws URISyntaxException {
+    WebServiceDriver webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+    WebServiceDriver webServiceDriver2 = new WebServiceDriver("http://www.google.com/");
+
+    WebServiceDriverManager driverManager = new WebServiceDriverManager(webServiceDriver, this.getTestObject());
+    driverManager.overrideDriver(webServiceDriver2);
+
+    Assert.assertEquals(driverManager.getWebServiceDriver().getBaseWebServiceAddress().toString(), "http://www.google.com/");
+  }
+
+  @Test(groups = TestCategories.WebService)
+  public void closeWebServiceDriverTest() throws URISyntaxException {
+    WebServiceDriver webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+    WebServiceDriverManager driverManager = new WebServiceDriverManager(webServiceDriver, this.getTestObject());
+
+    driverManager.close();
+    Assert.assertNull(driverManager.getBaseDriver(), "Expected Base Driver to be null.");
+  }
+
+  @Override protected void postSetupLogging() throws Exception {
+
```

Comment:
@jason-edstrom should this be deprecated?

Reply:
Good catch, although it doesn't matter too much in the Unit Test Class.

Reply:
He is extending off of BaseTest so the abstract paradigm is pulling that override through the inheritance chain.  Nothing wrong with it being there at the moment since the test would not compile without it.  I would mark with a FIXME at the most.  That inheritance will go away once all the child modules have broken away from the pattern.